pkg/services/ngalert/metrics/multi_org_alertmanager.go

```diff
@@ -28,5 +37,112 @@ func NewMultiOrgAlertmanagerMetrics(r prometheus.Registerer) *MultiOrgAlertmanag
+// Unlike metrics.Alertmanager they are not called within this codebase hence the need for direct collection.
+type AlertmanagerAggregatedMetrics struct {
+	registries *UserRegistries
+
+	// exported metrics, gathered from Alertmanager Silences
+	silencesGCDuration              *prometheus.Desc
+	silencesSnapshotDuration        *prometheus.Desc
+	silencesSnapshotSize            *prometheus.Desc
+	silencesQueriesTotal            *prometheus.Desc
+	silencesQueryErrorsTotal        *prometheus.Desc
+	silencesQueryDuration           *prometheus.Desc
+	silences                        *prometheus.Desc
+	silencesPropagatedMessagesTotal *prometheus.Desc
+}
+
+func NewAlertmanagerAggregatedMetrics(registries *UserRegistries) *AlertmanagerAggregatedMetrics {
+	aggregatedMetrics := &AlertmanagerAggregatedMetrics{
+		registries: registries,
+
+		silencesGCDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_gc_duration_seconds", Namespace, Subsystem),
+			"Duration of the last silence garbage collection cycle.",
+			nil, nil),
+		silencesSnapshotDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_snapshot_duration_seconds", Namespace, Subsystem),
+			"Duration of the last silence snapshot.",
+			nil, nil),
+		silencesSnapshotSize: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_snapshot_size_bytes", Namespace, Subsystem),
+			"Size of the last silence snapshot in bytes.",
+			nil, nil),
+		silencesQueriesTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_queries_total", Namespace, Subsystem),
+			"How many silence queries were received.",
+			nil, nil),
+		silencesQueryErrorsTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_query_errors_total", Namespace, Subsystem),
+			"How many silence received queries did not succeed.",
+			nil, nil),
+		silencesQueryDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_query_duration_seconds", Namespace, Subsystem),
+			"Duration of silence query evaluation.",
+			nil, nil),
+		silencesPropagatedMessagesTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences_gossip_messages_propagated_total", Namespace, Subsystem),
+			"Number of received gossip messages that have been further gossiped.",
+			nil, nil),
+		silences: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_silences", Namespace, Subsystem),
+			"How many silences by state.",
+			[]string{"org", "state"}, nil),
 	}
+
+	return aggregatedMetrics
+}
+
+func (a *AlertmanagerAggregatedMetrics) Describe(out chan<- *prometheus.Desc) {
+	out <- a.silencesGCDuration
+	out <- a.silencesSnapshotDuration
+	out <- a.silencesSnapshotSize
+	out <- a.silencesQueriesTotal
+	out <- a.silencesQueryErrorsTotal
+	out <- a.silencesQueryDuration
+	out <- a.silencesPropagatedMessagesTotal
+	out <- a.silences
+}
+
+func (a *AlertmanagerAggregatedMetrics) Collect(out chan<- prometheus.Metric) {
+	data := a.registries.BuildMetricFamiliesPerUser()
+
+	data.SendSumOfSummaries(out, a.silencesGCDuration, "alertmanager_silences_gc_duration_seconds")
```

Comment:
You can go read the function signature to get a gist of what's going but it gets hairy please quickly.

Of each `SendSum` function there's two kinds, one with a suffix of `PerUser` and one without the suffix. These allow us to control if you want to perform an aggregation (and effectively turn a metric that will be per user to per instance).

For example, say we have `alertmanager_silences_snapshot_size_bytes` and 3 Grafana tenants. With the per user metric we would get:

```
alertmanager_silences_snapshot_size_bytes{org="1"} 10
alertmanager_silences_snapshot_size_bytes{org="2"} 20
alertmanager_silences_snapshot_size_bytes{org="3"} 30
```
With the function that does not have the suffix we get:

```
alertmanager_silences_snapshot_size_bytes 50
```

This technique helps us reduce per-tenant cardinality. We did a pretty extensive analysis in Mimir back when we implemented this so I just copied the work that we did - that being said, there are probably changes that we'll make over time as we discover what kind of balance works for Grafana.